src/util/codec/datum.rs

```diff
@@ -299,23 +299,32 @@ impl Datum {
         Ok(b)
     }
 
-    /// into_string convert self into a string.
-    /// source function name is `ToString`.
-    pub fn into_string(self) -> Result<String> {
-        let s = match self {
+    pub fn as_string(&self) -> Result<String> {
```

Comment:
IMO it's better to use name `into_string`. For name convention in Rust, `as_xxx` is usually for returning a reference, such as, `as_bytes()`, `as_str()`, `as_ref()`, while `into_xxx` is for type conversion, such as, `into_bytes()`, `into_inner()`.